src/harness/cbsd_sas_harnes/sas_interface.py

```diff
@@ -0,0 +1,422 @@
+#!/usr/bin/env python
+# -- encoding: utf-8 --
+
+__author__ = "Francisco Benavides (francisco.benavides@federatedwireless.com)"
+"""
+    SAS Interface
+    Federated Wireless
+
+    SAS to User Protocol Technical Specification
+    WINNF-16-S-0016 SAS to CBSD v0.3.10, 12 July 2016
+
+    9.2     Message Transport
+            HTTP v1.1, Content-type: application/json
+            JSON objects shall use UTF-8 enconding
+            All requests from CBSD to SAS shall use the HTTP POST method
+
+    SAS - SAS Interface Technical Specification
+    WINNF-16-S-0096 v0.3.5, 20 September 2016
+
+    6.1     SAS Mutual Authentication and Communication Security
+            TLS v1.2 authentication
+    7.1     Message encoding
+            JSON objects UTF-8 encoding
+    7.2     Message Transport
+            HTTP 1.1, Content-type: application/json
+            GET (Pull) and POST (Push) methods
+"""
+
+
+# Imports
+
+import json
+import os
+import sys
+import ConfigParser
+import logging
+import re
+# PycURL
+import StringIO
+import urllib
+import urlparse
+import pycurl
+# Import OWN
+from sas_connector import SAS_Connector
+
+
+# Constants
+
+# Log format
+LOG_FORMAT = logging.Formatter(
+    '%(asctime)-15s - SAS Interface - '
+    '%(funcName)s - %(levelname)s - %(message)s')
+# SAS Interface Settings
+INI_FILE = 'tests/sas_settings.ini'
+INI_FILE_SECTION = 'HOST'
+INI_FILE_OPTIONS = {
+    'VERSION': None,
+    'HTTP': None,
+    'HTTPS': None,
+    'CA_CERT': None,
+    'CERT': None,
+    'KEY': None,
+    'KEYPASSWD': None}
+PULL = True
+PUSH = not(PULL)
+HTTP_TIMEOUT_SECS = 300
+# TLS 1.2 Ciphers
+SSL_CIPHER_LIST = ':'.join([
+    'RSA-AES128-GCM-SHA256',
+    'RSA-AES256-GCM-SHA384',
+    'ECDHE-ECDSA-AES128-GCM-SHA256',
+    'ECDHE-ECDSA-AES256-GCM-SHA384',
+    'ECDHE-RSA-AES128-GCM-SHA256'])
+
+
+# Classes
+
+
+class CConfigParser(ConfigParser.ConfigParser):
```

Comment:
I had modified it to interpolate options defined both within a section and between sections, since we are now using the new config based upon the agreement, I have discarded the iterpolation/read code and left it smaller, so as to preserv the option 'case format' which allows me to not to have to move more code than what is needed.